packages/blocks/src/store/actions.js

```diff
@@ -47,11 +55,13 @@ function isFunction( maybeFunc ) {
  * Takes the unprocessed block type data and applies all the existing filters for the registered block type.
  * Next, it validates all the settings and performs additional processing to the block type definition.
  *
- * @param {WPBlockType} blockType        Unprocessed block type settings.
- * @param {Object}      thunkArgs        Argument object for the thunk middleware.
- * @param {Function}    thunkArgs.select Function to select from the store.
+ * TODO Originally `thunkArgs.select` was typed as a function, but it is used as an object
  *
- * @return {WPBlockType | undefined} The block, if it has been successfully registered; otherwise `undefined`.
+ * @param {BlockType} blockType        Unprocessed block type settings.
+ * @param {Object}    thunkArgs        Argument object for the thunk middleware.
+ * @param {Object}    thunkArgs.select Function to select from the store.
+ *
+ * @return {BlockType | undefined} The block, if it has been successfully registered; otherwise `undefined`.
```

Comment:
Originally `thunkArgs.select` was typed as a function, but it is used as an object